### DISCUSSION TITLE
Query dataview

### DISCUSSION THREAD

**Discussion by gijs-epping** [coddingtonbear/obsidian-local-rest-api#8] [Ideas]
Hi,

Cool plugin. Using it together with n8n desktop, awesome stuff. 

Question though, would it be possible to execute dataview queries?

**Comment by coddingtonbear** [comment]
Nothing currently exists for that, but looking at the dataview repo, it does look like there's a usable public API for that: https://github.com/blacksmithgu/obsidian-dataview/blob/master/src/typings/api.d.ts#L79.

So, it looks like it'd probably be pretty easy to add a new query format to the [`/search/` API endpoint](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Search/post_search_) supporting dataview queries.

**Comment by coddingtonbear** [comment]
I spent a couple hours looking into this this weekend, and it's apparently not quite as easy as it looked.  Although some parts of the dataview API are exposed, the query parser is not: https://github.com/blacksmithgu/obsidian-dataview/issues/374.  I've pushed up the in-flight changes in the branch `dataview_queries` in case that changes someday in the future and somebody would like to try getting this across the finish line, but at the moment it appears to not be an option.

**Comment by coddingtonbear** [comment]
Actually: apparently since my writing that previous message, this is actually possible now: https://github.com/blacksmithgu/obsidian-dataview/issues/374#issuecomment-1152040331.

**Comment by coddingtonbear** [comment]
Support for Dataview DQL queries was added in v1.4.0.

**Comment by coddingtonbear** [comment]
& removed in v4.0.0
